src/compose.ts

```diff
@@ -274,3 +298,12 @@ function requestUpdate(composer: Compose) {
     processChanges(composer);
   });
 }
+
+function determineActivationStrategy(composer: Compose) {
+  let activationStrategy = composer.activationStrategy;
+  const vm = composer.currentViewModel;
+  if (vm && typeof vm.determineActivationStrategy === 'function') {
+    activationStrategy = vm.determineActivationStrategy();
```

Comment:
What if devs make a typo in `determineActivationStrategy` return? Do we normalize it or throw? At the moment, it doesn't handle it at all. I'd love to see a test for whatever the solution we come up with for this.
Thoughts?
cc @EisenbergEffect @fkleuver

Reply:
For one, it is expected that devs use the `export`ed enum. However, if they make any mistake, then it will not be the `ActivationStrategy.Replace`, and the default strategy will come to play. Even if there comes 3rd strategy in future, it should not be a problem as long as it is matched against the enum values; and if it a no-match the default strategy will be used. Is an explicit error required in this case?

Reply:
Does the router throw on unknown strategy?

Reply:
No it does not, as it check very differently 
https://github.com/aurelia/router/blob/893b768f01aea842ee57db4222e66aa572f24404/src/navigation-plan.ts#L123

https://github.com/aurelia/router/blob/893b768f01aea842ee57db4222e66aa572f24404/src/navigation-plan.ts#L142

@Sayan751 I forgot this `&& determineActivationStrategy(composer) !== ActivationStrategy.Replace` part when reviewing this bit. So I think we are good. But maybe let's add a test or two returning the wrong thing to solidify the logic here?

Reply:
Added negative tests